tests/test_giturl.py

```diff
@@ -0,0 +1,65 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from unittest import mock
+
+from fetchcode import giturl
+
+@mock.patch('os.system')
+def test_fetch_with_https_url_returns_a_response(mock_os):
+    url = 'https://github.com/TG1999/fetchcode.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response.url_scheme == 'https'
+        assert response.domain == 'github.com'
+
+
+@mock.patch('os.system')
+def test_fetch_with_git_url_returns_a_response(mock_os):
+    url = 'git://github.com/jamesor/mongoose-versioner'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response.url_scheme == 'git'
+        assert response.domain == 'github.com'
+
+
+@mock.patch('os.system')
+def test_fetch_with_git_ssh_url_returns_a_response(mock_os):
+    url = 'git+ssh://github.com/bcoe/thumbd.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response.url_scheme == 'git+ssh'
+        assert response.domain == 'github.com'
+
+
+@mock.patch('os.system')
+def test_fetch_with_git_https_url_returns_a_response(mock_os):
+    url = 'git+https://github.com/Empeeric/i18n-node'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response.url_scheme == 'git+https'
+        assert response.domain == 'github.com'
+
+
+@mock.patch('os.system')
+def test_fetch_with_ssh_url_returns_a_response(mock_os):
+    url = 'ssh://git@github.com:EastCloud/node-websockets.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response.url_scheme == 'ssh'
+        assert response.domain == 'git@github.com:EastCloud'
```

Comment:
Shouldn't the domain be `github.com` as well for this case?

Reply:
Okay, I am getting you, can you suggest me some way in which I can parse the domain for ssh based URLs currently I am using urllib.parse which is giving me this kind of domain name, what approach should I use for it or should I handle ssh case differently and do some string manipulation on my own

Reply:
@TG1999 Well, the first step would be to look at the urllib docs to see exactly how things like this address get parsed. If there is nothing there, then we probably want to handle this as a special case. I havent looked in depth at the docs myself, so I do not know the exact way to approach this.

Reply:
Hi @majurg , I have handled it separately now all test cases are working fine too, good to go?